module/motion/Servos/src/Servos.hpp

```diff
@@ -60,11 +60,57 @@ namespace module::motion {
 
                     // Runs an emit for each servo
                     NUClear::util::unpack((emit<Task>(std::make_unique<Elements>(
-                                               group.servos[utility::motion::ServoMap<Elements>::value])),
+                                               group.servos.at(utility::motion::ServoMap<Elements>::value))),
                                            0)...);
                 });
         }
 
+        /// @brief Keeps track of which Group in the Sequence in add_sequence_provider should be emitted next
+        /// @tparam Sequence Used to have unique counts for each Sequence Provider.
+        template <typename Sequence>
+        struct Count {
+            long unsigned int count = 0;
+        };
+
+        /// @brief Creates a reaction that takes a vector of servo wrapper tasks (eg LeftLegs) and emits each servo
+        /// wrapper task after the previous one is Done. Emits Done when the last group is Done.
+        /// @tparam Sequence is a vector of servo wrapper tasks (eg LeftLegSequence)
+        /// @tparam Group is a servo wrapper task (eg LeftLeg)
+        template <typename Sequence, typename Group>
+        void add_sequence_provider() {
+            // Message to keep track of which position in the vector is to be emitted
+            // Make an initial count message
+            emit<Scope::DIRECT>(std::make_unique<Count<Sequence>>(0));
+
+            on<Provide<Sequence>, Needs<Group>, With<Count<Sequence>>>().then(
+                [this](const Sequence& sequence, const RunInfo& info, const Count<Sequence>& count) {
+                    // If the user gave us nothing then we are done
+                    if (sequence.pack.empty()) {
+                        emit<Task>(std::make_unique<Done>());
+                    }
+                    // If this is a new task, run the first pack of servos and increment the counter
+                    else if (info.run_reason == RunInfo::RunReason::NEW_TASK) {
+                        emit<Task>(std::make_unique<Group>(sequence.pack[0]));
+                        emit<Scope::DIRECT>(std::make_unique<Count<Sequence>>(1));
+                    }
+                    // If the subtask is done, we are done if it is the last servo pack, otherwise use the count to
+                    // determine the current pack to emit
+                    else if (info.run_reason == RunInfo::RunReason::SUBTASK_DONE) {
+                        if (sequence.pack.size() < count.count) {
```

Comment:
This reads backwards to me
shouldn't it be `count.count < sequence.pack.size()` emit the next frame else emit done?
More like a loop counter